cap/modules/deposit/serializers/json.py

```diff
@@ -41,5 +41,10 @@ def preprocess_record(self, pid, record, links_factory=None, **kwargs):
         result = super().preprocess_record(
             pid, record, links_factory=deposit_links_factory)
 
-        result['deposit'] = CAPDeposit.get_record(pid.object_uuid)
+        deposit = CAPDeposit.get_record(pid.object_uuid)
+        result['deposit'] = deposit
+
+        # add bucket id for fetching files later
+        if deposit.files:
```

Comment:
deposit is a record instance and will be a dict, If yes then having `.get` would be better.

```suggestion
        if deposit.get('files'):
```